core/shared/src/main/scala/zio/Scope.scala

```diff
@@ -242,9 +242,9 @@ object Scope {
       ZIO.suspendSucceed {
         var b = null.asInstanceOf[UIO[B]]
         while (b eq null) {
-          val current        = ref.get()
-          val (value, state) = f(current)
-          if (ref.compareAndSet(current, state)) b = value
+          val current = ref.get()
+          val kv      = f(current)
+          if (ref.compareAndSet(current, kv._2)) b = kv._1
```

Comment:
Suggestion to keep the nice names. It's easier to read
```suggestion
          val kv      = f(current)
          val value = kv._1
          val state = kv._2
          if (ref.compareAndSet(current, state)) b = value
```